core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -192,6 +195,7 @@ public Node(Settings preparedSettings) {
         this(InternalSettingsPreparer.prepareEnvironment(preparedSettings, null), Collections.<Class<? extends Plugin>>emptyList());
     }
 
+    @SuppressWarnings("unchecked")
```

Comment:
Actually this was unnecessary, I just removed it. Yes there is a warning, but it is already suppressed (core has unchecked removed). And this will go away once we get rid of guice anyways, it is really a bug on the crazy generics for its bind methods.